demo_nodes_cpp/src/logging/use_logger_service.cpp

```diff
@@ -0,0 +1,207 @@
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "rcl_interfaces/srv/get_logger_levels.hpp"
+#include "rcl_interfaces/srv/set_logger_levels.hpp"
+
+using namespace std::chrono_literals;
+
+// This demo program shows how to enable logger service and control logger level via logger service.
+// Class LoggerServiceNode enable logger service and create a subscription. The callback of
+// subscription output received message by different log functions.
+// Class TestNode can set/get logger level of LoggerServiceNode and send message to it.
+
+class LoggerServiceNode : public rclcpp::Node
+{
+public:
+  explicit LoggerServiceNode(const std::string & node_name)
+  : Node(node_name, rclcpp::NodeOptions().enable_logger_service(true))
+  {
+    auto callback = [this](std_msgs::msg::String::ConstSharedPtr msg)-> void {
+        RCLCPP_DEBUG(this->get_logger(), "%s log with DEBUG logger level.", msg->data.c_str());
+        RCLCPP_INFO(this->get_logger(), "%s with INFO logger level.", msg->data.c_str());
+        RCLCPP_WARN(this->get_logger(), "%s with WARN logger level.", msg->data.c_str());
+        RCLCPP_ERROR(this->get_logger(), "%s with ERROR logger level.", msg->data.c_str());
+      };
+
+    sub_ = this->create_subscription<std_msgs::msg::String>("output", 10, callback);
+  }
+
+  ~LoggerServiceNode() = default;
+
+private:
+  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;
+};
+
+class TestNode : public rclcpp::Node
+{
+public:
+  explicit TestNode(const std::string & remote_node_name)
+  : Node("TestNode"),
+    remote_node_name_(remote_node_name)
+  {
+    pub_ = this->create_publisher<std_msgs::msg::String>("output", 10);
+    logger_set_client_ = this->create_client<rcl_interfaces::srv::SetLoggerLevels>(
+      remote_node_name + "/set_logger_levels");
+    logger_get_client_ = this->create_client<rcl_interfaces::srv::GetLoggerLevels>(
+      remote_node_name + "/get_logger_levels");
+  }
+
+  ~TestNode() = default;
+
+  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr get_pub()
+  {
+    return pub_;
+  }
+
+  bool set_logger_level_on_remote_node(
+    rclcpp::Logger::Level logger_level)
+  {
+    if (!logger_set_client_->wait_for_service(2s)) {
+      return false;
+    }
+
+    auto request = std::make_shared<rcl_interfaces::srv::SetLoggerLevels::Request>();
+    auto set_logger_level = rcl_interfaces::msg::LoggerLevel();
+    set_logger_level.name = remote_node_name_;
+    set_logger_level.level = static_cast<uint8_t>(logger_level);
+    request->levels.emplace_back(set_logger_level);
+
+    auto result = logger_set_client_->async_send_request(request);
+
+    if (rclcpp::spin_until_future_complete(this->shared_from_this(), result) !=
+      rclcpp::FutureReturnCode::SUCCESS)
+    {
+      return false;
+    }
+
+    auto ret_result = result.get();
+    if (!ret_result->results[0].successful) {
+      return false;
```

Comment:
print outs the reason would be good for demonstration code?

```suggestion
      RCLCPP_ERROR(this->get_logger(), "Failed to change logger level: %s",
        ret_result->results[0].reason.c_str());
      return false;
```